app/Config/Cookie.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Config;
+
+use CodeIgniter\Config\BaseConfig;
+use DateTimeInterface;
+
+class Cookie extends BaseConfig
+{
+	/**
+	 * --------------------------------------------------------------------------
+	 * Cookie Prefix
+	 * --------------------------------------------------------------------------
+	 *
+	 * Set a cookie name prefix if you need to avoid collisions.
+	 *
+	 * @var string
+	 */
+	public $prefix = '';
+
+	/**
+	 * --------------------------------------------------------------------------
+	 * Cookie Expires Timestamp
+	 * --------------------------------------------------------------------------
+	 *
+	 * Default expires timestamp for cookies. Setting this to `0` will mean the
+	 * cookie will not have the `Expires` attribute and will behave as a session
+	 * cookie.
+	 *
+	 * @var DateTimeInterface|integer|string
+	 */
+	public $expires = 0;
+
+	/**
+	 * --------------------------------------------------------------------------
+	 * Cookie Path
+	 * --------------------------------------------------------------------------
+	 *
+	 * Typically will be a forward slash.
+	 *
+	 * @var string
+	 */
+	public $path = '/';
+
+	/**
+	 * --------------------------------------------------------------------------
+	 * Cookie Domain
+	 * --------------------------------------------------------------------------
+	 *
+	 * Set to `.your-domain.com` for site-wide cookies.
+	 *
+	 * @var string
+	 */
+	public $domain = '';
+
+	/**
+	 * --------------------------------------------------------------------------
+	 * Cookie Secure
+	 * --------------------------------------------------------------------------
+	 *
+	 * Cookie will only be set if a secure HTTPS connection exists.
+	 *
+	 * @var boolean
+	 */
+	public $secure = false;
+
+	/**
+	 * --------------------------------------------------------------------------
+	 * Cookie HTTPOnly
+	 * --------------------------------------------------------------------------
+	 *
+	 * Cookie will only be accessible via HTTP(S) (no JavaScript).
+	 *
+	 * @var boolean
+	 */
+	public $httponly = true;
+
+	/**
+	 * --------------------------------- -----------------------------------------
+	 * Cookie SameSite
+	 * --------------------------------------------------------------------------
+	 *
+	 * Configure cookie SameSite setting. Allowed values are:
+	 * - None
+	 * - Lax
+	 * - Strict
+	 * - ''
+	 *
+	 * Alternatively, you can use the constant names:
+	 * - `Cookie::SAMESITE_NONE`
+	 * - `Cookie::SAMESITE_LAX`
+	 * - `Cookie::SAMESITE_STRICT`
+	 *
+	 * Defaults to `Lax` for compatibility with modern browsers. Setting `''`
+	 * (empty string) means default SameSite attribute set by browsers (`Lax`)
+	 * will be set on cookies. If set to `None`, `$cookieSecure` must also be set.
```

Comment:
$cookieSecure change to $secure

Reply:
shoud we force set secure to true when developer uses None a samesite attribute?

Reply:
That is in the cookie spec. None samesite should have the Secure attribute. Otherwise, the cookie is rejected.

Reply:
what is this happend?

if (strtolower($samesite) === self::SAMESITE_NONE && ! $secure)
{
	//throw CookieException::forInvalidSameSiteNone();
	self::setDefaults(['secure' => true];
}

Reply:
That would be condoning the developer's mistake. What sets apart CI4 from CI3 is that we throw exceptions for coding mistakes rather than the framework correcting them and have the dev with no idea that they have a mistake.